x-pack/plugin/inference/src/yamlRestTest/resources/rest-api-spec/test/inference/10_semantic_text_inference.yml

```diff
@@ -426,14 +396,13 @@ setup:
         index: test-copy-to-index
         id: doc_1
 
-  - match:  { _source.inference_field: "inference test" }
-  - length: { _source._inference.inference_field.chunks: 3 }
-  - exists: _source._inference.inference_field.chunks.0.inference
-  - exists: _source._inference.inference_field.chunks.0.text
-  - exists: _source._inference.inference_field.chunks.1.inference
-  - exists: _source._inference.inference_field.chunks.1.text
-  - exists: _source._inference.inference_field.chunks.2.inference
-  - exists: _source._inference.inference_field.chunks.2.text
+  - match:  { _source.inference_field.raw: "inference test" }
+  - match:  { _source.inference_field.inference.chunks.0.text: "another copy_to inference test" }
+  - exists:   _source.inference_field.inference.chunks.0.embeddings
+  - match:  { _source.inference_field.inference.chunks.1.text: "inference test" }
+  - exists:   _source.inference_field.inference.chunks.1.embeddings
+  - match:  { _source.inference_field.inference.chunks.2.text: "copy_to inference test" }
+  - exists:   _source.inference_field.inference.chunks.2.embeddings
```

Comment:
Can we rely on the order of inference results always being the same?

Also, can we use `length` to check that there are three chunks?

Reply:
> Can we rely on the order of inference results always being the same?

Yes, the source fields are sorted consistently and we preserve the order in the inference response.

I'll add the length check.

Reply:
https://github.com/elastic/elasticsearch/pull/107065/commits/03ab6ccf46f4987e57c5f2881c05f4d7f8f51e57